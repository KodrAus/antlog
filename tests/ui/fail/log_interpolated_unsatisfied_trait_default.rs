#![feature(stmt_expr_attributes, proc_macro_hygiene)]

#[macro_use]
extern crate emit;

// Does not implement `Display`
struct Input;

fn main() {
    emit!("Text \"and\" {a: Input} and more");
}
