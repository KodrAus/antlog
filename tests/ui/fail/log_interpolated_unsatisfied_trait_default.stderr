error[E0277]: `Input` doesn't implement `std::fmt::Display`
  --> $DIR/log_interpolated_unsatisfied_trait_default.rs:10:26
   |
10 |     info!("Text \"and\" {a: Input} and more");
   |                          ^^^^^^^^ `Input` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `Input`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: required because of the requirements on the impl of `emit::emit_rt::capture::Capture<emit::emit_rt::capture::CaptureDisplay>` for `Input`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
